### PR TITLE
[codex] Support operation result response headers

### PR DESCRIPTION
**Pull request by hughhan1**
## Summary

Carries operation response headers through the shared provider protocol and SDK runtimes instead of treating `Content-Type` as the only response header that can reach HTTP clients.

The provider `OperationResult` now serializes multi-value response headers, and the Go, Python, Rust, and TypeScript SDKs expose the same header shape from authored operation responses and app invocation clients.

This lets hosted HTTP operations return real redirect and response metadata such as `302` plus `Location`, repeated `Set-Cookie`, custom headers, and non-JSON `Content-Type` without adding per-header branches.

## Interfaces

```proto
message OperationResult {
  int32 status = 1;
  string body = 2;
  map<string, StringList> headers = 3;
}
```

```ts
response(302, {}, { Location: "/next" });
ok({ id: "created" }, { Location: "/items/created" });
```

```go
type Response[T any] struct {
  Status  int
  Headers http.Header
  Body    T
}
```

```python
return Response(
    status=302,
    body={},
    headers={"Location": "/next"},
)
```

```rust
ok(body).with_header("Location", "/next")
```

## Runtime

Gestaltd now copies every header from `core.OperationResult.Headers` to the HTTP response writer. The writer does not synthesize `Content-Type`; SDK operation serializers attach `Content-Type: application/json` when they JSON-encode a response and the handler did not provide a content type.

The app access and provider adapters preserve headers when converting between core results and protobuf results, and SDK app clients preserve headers from downstream app invocations.

The old response-writer `Content-Type` branch is deleted; redirects and other response metadata use the same generic header path as every other header.

### CODE REVIEW

File: sdk/rust/src/app_access.rs [hunk 1]
```diff
@@ -464,3 +467,12 @@ fn serializable_to_optional_struct<T: Serialize>(
...
 }
+
+fn string_lists(
+    values: &BTreeMap<String, crate::generated::v1::StringList>,
+) -> BTreeMap<String, Vec<String>> {
+    values
+        .iter()
+        .map(|(key, value)| (key.clone(), value.values.clone()))
+        .collect()
+}
```
[the remark this answers is not in the document]
Comment (hughhan1) [reply sdk/rust/src/app_access.rs]: Fixed in 7b2a535d8 by moving the Rust proto `StringList` map conversion into the shared crate-local protocol helper and reusing it from both app access and provider server paths.

File: sdk/python/gestalt/_app_access.py [hunk 2]
```diff
@@ -327,6 +325,24 @@ def _with_app_relay_token(channel: grpc.Channel, token: str) -> grpc.Channel:
     return grpc.intercept_channel(channel, interceptor)
 
 
+def _string_lists_from_proto_map(values: Any) -> dict[str, list[str]]:
+    return {
+        str(key): list(getattr(value, "values", ()))
+        for key, value in dict(values or {}).items()
+    }
```
[the remark this answers is not in the document]
Comment (hughhan1) [reply sdk/python/gestalt/_app_access.py]: Fixed in 5cc60ce07 by moving the Python proto `StringList` map conversion into the shared `_protocol` helper and reusing it from both app access and runtime paths.

File: gestaltd/services/appaccess/app_server.go [hunk 3]
```diff
@@ -146,11 +147,25 @@ func (s *AppServer) InvokeGraphQL(ctx context.Context, req *proto.AppInvokeGraph
...
+		return nil
+	}
+	out := make(map[string]*proto.StringList, len(values))
+	for key, item := range values {
+		copied := make([]string, len(item))
+		copy(copied, item)
+		out[key] = &proto.StringList{Values: copied}
+	}
+	return out
+}
```
[the remark this answers is not in the document]
Comment (hughhan1) [reply gestaltd/services/appaccess/app_server.go]: Fixed in 62d1017fa by moving the Go proto `StringList` map conversion helpers into `services/internal/protoutil` and reusing them from both app access and apps paths.

File: gestaltd/internal/server/response.go [hunk 4]
```diff
@@ -37,14 +37,13 @@ func writeOperationResult(w http.ResponseWriter, result *core.OperationResult) {
...
-		if ct := result.Headers.Get("Content-Type"); ct != "" {
-			contentType = ct
+		for name, values := range result.Headers {
+			for _, value := range values {
+				w.Header().Add(name, value)
+			}
 		}
 	}
-
-	w.Header().Set("Content-Type", contentType)
```
[the remark this answers is not in the document]
Comment (hughhan1) [reply gestaltd/internal/server/response.go:47]: Fixed in e3c4857ed by keeping the gestaltd writer generic but moving the JSON content type default into SDK operation-result serialization. Go/Python/Rust/TypeScript now attach `Content-Type: application/json` when they JSON-encode an operation result unless the handler supplied a content type, and the focused SDK/server checks pass locally.

File: sdk/python/gestalt/_runtime.py [hunk 5]
```diff
@@ -1608,7 +1609,11 @@ def Execute(self, request: Any, _context: Any) -> Any:
                 status = HTTPStatus.INTERNAL_SERVER_ERROR
                 body = json_body({"error": INTERNAL_ERROR_MESSAGE})
                 return app_pb2.OperationResult(status=status, body=body)
-            return app_pb2.OperationResult(status=result.status, body=result.body)
```
[the remark this answers is not in the document]
Comment (hughhan1) [reply sdk/python/gestalt/_runtime.py:1611]: Fixed in 524d224a5 by adding `Content-Type: application/json` to the Python runtime catch-all `Execute` error result. I also added a runtime test assertion for the error-path header and reran the Python type, lint, runtime, app, and app transport tests locally.

File: sdk/go/router.go [hunk 6]
```diff
@@ -332,6 +333,17 @@ func normalizeMethod(method string) string {
...
+func jsonResponseHeaders(headers http.Header) http.Header {
+	out := headers.Clone()
+	if out == nil {
+		out = http.Header{}
+	}
+	if out.Get("Content-Type") == "" {
+		out.Set("Content-Type", "application/json")
+	}
+	return out
+}
```
[the remark this answers is not in the document]
Comment (hughhan1) [reply sdk/go/router.go:345]: Fixed in e4922b536 by changing Go `jsonResponseHeaders` to detect existing `Content-Type` headers with a case-insensitive key scan. I also added a router regression test for a non-canonical `content-type` key so we do not add a duplicate JSON content type.
